src/Files.App/Helpers/Layout/AdaptiveLayoutHelpers.cs

```diff
@@ -36,49 +32,31 @@ public static void ApplyAdaptativeLayout(LayoutPreferencesManager folderSettings
 			}
 		}
 
-		private static Layouts GetAdaptiveLayout(string path, IList<ListedItem> filesAndFolders)
+		private static Layouts GetAdaptiveLayout(IList<ListedItem> filesAndFolders)
 		{
-			var pathLayout = GetPathLayout(path);
+			var pathLayout = GetPathLayout();
 			if (pathLayout is not Layouts.None)
 				return pathLayout;
 
 			return GetContentLayout(filesAndFolders);
 		}
 
-		private static Layouts GetPathLayout(string path)
+		private static Layouts GetPathLayout()
 		{
-			var iniPath = IO.Path.Combine(path, "desktop.ini");
-
-			var iniContents = Win32Helper.ReadStringFromFile(iniPath)?.Trim();
-			if (string.IsNullOrEmpty(iniContents))
-				return Layouts.None;
-
-			var parser = new IniParser.Parser.IniDataParser();
-			parser.Configuration.ThrowExceptionsOnError = false;
-			var data = parser.Parse(iniContents);
-			if (data is null)
-				return Layouts.None;
+			var desktopIni = ContentPageContext.ShellPage!.ShellViewModel.DesktopIni;
 
-			var viewModeSection = data.Sections.FirstOrDefault(IsViewState);
-			if (viewModeSection is null)
-				return Layouts.None;
+			var viewStateSection = desktopIni.FirstOrDefault(x => x.SectionName == "ViewState");
+			if (viewStateSection is null)
+				return Layouts.Detail;
 
-			var folderTypeKey = viewModeSection.Keys.FirstOrDefault(IsFolderType);
-			if (folderTypeKey is null)
-				return Layouts.None;
+			var viewMode = viewStateSection.Parameters?.FirstOrDefault(x => x.Key == "Mode").Value;
```

Comment:
```suggestion
			var viewMode = viewStateSection.Parameters.FirstOrDefault(x => x.Key == "Mode").Value;
```
We can now replace `Parameters?` to `Parameters`.

Reply:
I pointed out one, but there are others that need to be fixed as well.

Reply:
I pushed a commit to resolve